src/devtools/client/debugger/src/components/SecondaryPanes/ObjectPreviewSuspenseCacheAdapter.tsx

```diff
@@ -21,7 +36,9 @@ export default function ObjectPreviewSuspenseCacheAdapter() {
       if (objects) {
         // Be sure to clone object data before pre-caching it.
         // Otherwise ValueFronts might deeply mutate it and change its structure.
-        preCacheObjects(pauseId, JSON.parse(JSON.stringify(objects)));
+        // Because object data may contain circular references, we need to also filter out empty entries.
+        const cloned = JSON.parse(JSON.stringify(objects, createReplacer())).filter(Boolean);
+        preCacheObjects(pauseId, cloned);
```

Comment:
This is gross. I wish we didn't have to be so heavy handed, but `ValueFront` sucks.

Reply:
This is one of the grossest things i've seen in awhile. Can't remember the last time I used the 2nd param to `stringify`

Reply:
I can 🤓 

https://github.com/replayio/devtools/blob/127c09c5dfb1c2989d1d2522bf2bc7b83cd4bcc3/packages/shared/client/encoder.ts#L13-L27

Reply:
oh god - you are the replacer!